mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -3312,6 +3312,7 @@ auto ConversionTarget::getOpInfo(OperationName op) const
   return std::nullopt;
 }
 
+#if MLIR_ENABLE_PDL
```

Comment:
Can we extract all of this in a separate .cpp and handle this with separate CMakeLibraries?

Reply:
This one is trickier as the parts here depend on ConversionPatternRewriterImpl, so would result in weird library split.